testing/testing_index.adoc

```diff
@@ -0,0 +1,97 @@
+// vim: set syntax=asciidoc:
+[[Testing]]
+== Testing
+:data-uri:
+:toc:
+:toclevels 4:
+:homepage https://github.com/projectatomic/container-best-practices:
+
+=== Purpose of the testing
+
+Container images generally consist of distro packages and some scripts that help to start the container properly. For example in case of MariaDB database, it is set of RPMs with mariadb-server that provides the main functionality, but we need some scripts that handle proper database stack initialization, changes basic configuration based on the container run options etc.
+
+If we want to test basic functionality of the container, we do not have to test the RPM functionality, because it should have already be done during RPM develpment process. We need to focus on testing the additional scripts instead and test the API of the container -- whether it works as it is described. Again, in case of the MariaDB database, we do not run MariaDB unit test in the container, but we focus on whether the database is initialized properly, whether configuration is changed after proper options are set on container run command etc.
```

Comment:
If we want to test the basic functionality of the container, we do not have to test the RPM functionality.  The benefit of using the distribution packaging is that we know testing has already been done during the RPM develpment process.  Instead, we need to focus on testing the added scripts instead and the API of the container.  The goal is to determine if it works as described. For example, a MariaDB database container, we do not run the MariaDB unit tests in the container.  Instead we focus on whether the database is initialized, configured, and responds to commands properly.